Добавлены тестовые сценарии

hebert928 opened the pull request:
1. Выпадающий список в разделе «Вопросы о важном». 
2. Заказ самоката.

hebert928 committed:
add pom.xml

hebert928 committed:
Добавлены тестовые сценарии:
1. Выпадающий список в разделе «Вопросы о важном».
2. Заказ самоката.

pomegranate282922 commented on src/main/java/org/example/Main.java:
```diff
@@ -0,0 +1,5 @@
+package org.example;
+
+public class Main {
```
Это нужно удалить

pomegranate282922 commented on src/test/java/pages/HomePageTest.java:
```diff
@@ -0,0 +1,50 @@
+    @Parameterized.Parameters
+    public static Object[] [] getTestData() {
+        return new Object[][] {
+                {"accordion__heading-0", "Сутки — 400 рублей. Оплата курьеру — наличными или картой."},
```
Локаторов в тестовом классе быть не должно. Это нарушает Page Object Model.
Нужно исправить

pomegranate282922 commented on src/test/java/pages/HomePageTest.java:
```diff
@@ -0,0 +1,50 @@
+                {"accordion__heading-3", "Только начиная с завтрашнего дня. Но скоро станем расторопнее."},
+                {"accordion__heading-7", "Да, обязательно. Всем самокатов! И Москве, и Московской области."},
+        };
+    }
```
Нужно сделать все 8 проверок 
Нужно исправить

pomegranate282922 commented on src/test/java/pages/HomePageTest.java:
```diff
@@ -0,0 +1,50 @@
+    @Test
+    public void checkedFAQList(){
+        driver = new ChromeDriver();
+        driver.get("https://qa-scooter.praktikum-services.ru/");
```
Урлы лучше вынести как константы и переиспользовать  
Можно улучшить

pomegranate282922 commented on src/test/java/pages/OrderPageTest.java:
```diff
@@ -0,0 +1,112 @@
+    }
+    @Test
+    public void firstCheckMakeOrder() {
+        driver = new ChromeDriver();
```
Общие начала и концы тестов лучше вынести в before и after 
Также удобно использовать общий класс предок 
Можно улучшить

pomegranate282922 commented on src/test/java/pages/OrderPageTest.java:
```diff
@@ -0,0 +1,112 @@
+        objOrderPage.fillAboutLeaseForm(date, period, color, comment);
+        objOrderPage.clickYesButton();
+        boolean actual = objOrderPage.checkOrderPopupEnabled();
+        assertEquals(true, actual);
```
assertTrue

pomegranate282922 commented on src/test/java/pages/OrderPageTest.java:
```diff
@@ -0,0 +1,112 @@
+        objOrderPage.clickYesButton();
+        boolean actual = objOrderPage.checkOrderPopupEnabled();
+        assertEquals(true, actual);
+    }
```
Здесь нужно немного изменить логику теста, чтобы избежать ненужного дублирования того, что уже протестировано.
 Первый флоу ( верхняя кнопка) - тестируем минимум двумя тестовыми наборами до появления надписи Заказ оформлен.
 Второй флоу ( нижняя кнопка) - тестируем до появления формы ввода параметров заказа.
 Нужно исправить

pomegranate282922 commented on src/test/java/pages/HomePageTest.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    @Parameterized.Parameters
+    public static Object[] [] getTestData() {
```
Отлично что применена параметризация

pomegranate282922 reviewed (commented):
Хорошая работа! Тесты реализованы, применена параметризация.
Замечания:
1. Убрать локаторы из тестовых классов 
2. Чуть изменить логику тестов на создание заказа 
3. Удалить ненужный код

hebert928 committed:
Внесенные изменения:
1. Удалён ненужный класс Main.
2. Убраны локаторы из тестовых данных.
3. Добавлены недостающие тестовые данные в HomePageTest.
4. Urls вынесены в константы.
5. Создан родительский класс PageTest.
6. assertEquals заменён на assertTrue.
7. Изменён тест на вторую кнопку "Заказать".

hebert928 committed:
Удалены неиспользуемые импорты, локатор и комментарий.

pomegranate282922 commented on src/main/java/pages/HomePage.java:
```diff
@@ -0,0 +1,53 @@
+public class HomePage {
+
+    private WebDriver driver;
+    private final String URL = "https://qa-scooter.praktikum-services.ru";
```
Static final необходимы, иначе именуем url 
Но это константа, так что первый вариант 
Нужно исправить

pomegranate282922 commented on src/main/java/pages/OrderPage.java:
```diff
@@ -0,0 +1,109 @@
+
+    private WebDriver driver;
+
+    private final String URL = "https://qa-scooter.praktikum-services.ru/order";
```
Статик 
Нужно исправить

pomegranate282922 commented on src/test/java/pages/HomePageTest.java:
```diff
@@ -0,0 +1,43 @@
+@RunWith(Parameterized.class)
+public class HomePageTest extends PageTest {
+    private final int index;
+    private final String checkedText;
```
Обычно это называется expected

pomegranate282922 commented on src/test/java/pages/OrderPageTest.java:
```diff
@@ -0,0 +1,96 @@
+    }
+
+    @Test
+    public void goToOrderPage() {
```
Этот тест повторится дважды так как лежит в параметризованном классе.
Можно улучшить

pomegranate282922 commented on src/test/java/pages/PageTest.java:
```diff
@@ -0,0 +1,20 @@
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+public class PageTest {
```
Обычно называют BaseTest

pomegranate282922 reviewed (commented):
Все отлично, кроме нейминга- с ним давай разберемся. 
Статичная константа static final именуется заглавными буквами. Остальные - camelCase

hebert928 committed:
Исправлены критические замечания

pomegranate282922 reviewed (approved):
Отлично, зачет!
